### ISSUE TITLE
Will an abi be the same if the same files are input?  I'm in a dilema

### ISSUE THREAD

**Issue by negaunee042** [hashlips-lab/nft-erc721-collection#114]
I deployed a contract on mainnet, and the abi didn't save in vscode.  I tried it again since gas was pretty low, but still expensive overall if it's a loss.  Same thing happened, no abi.  I tried compiling it myself without luck.  Not sure if it's salvageable or not.  I need some sleep tho.  Thx.

**Comment by liarco** [comment]
The ABI is compiled inside the `/smart-contract/artifacts`. Let me know if you can find it...

**Comment by negaunee042** [comment]
I went to verify again and it's an issue saying unable to find undefined reading "reentrancy guard".  I removed reentrancy cuz when I did a test in remix it said there was a constructor issue involving that in my contract (I think because I remove the withdrawal function with a 3-way splitter, not using the .call method.   I removed the reentrancy artifacts and the reentrancy from the json files but still getting a warning.  The json files never updated from my tests I believe.  I removed reentrancy from the abi and json-cache-file, but still can't figure out where the reentrancy is getting called to verify since it's been removed since last 2 deploys.   Thanks for your time & help. Getting this code;
Error HH700: Artifact for contract "@openzeppelin/contracts/security/ReentrancyGuard.sol:ReentrancyGuard" not found.

**Comment by negaunee042** [comment]
> Hi @negaunee042, the ABI is compiled inside the `/smart-contract/artifacts`. Let me know if you can find it...

The only ones there are from old tests.  I've tried a couple dozen times.   I tried compiling on remix, i tried manually with each contract and no luck.  I suspect changing the contract name didn't update an artifact or something similar.  I had that happen in testing 2 out of maybe 15 times.  Any ideas or think it's a lost cause?

**Comment by negaunee042** [comment]
I think my main issue is I removed Reentrancy because I replaced the withdraw function with a 3way splitter that doesn't make a "call".  So remix gave me constructor warnings for having reentrancy and not using it.   Maybe the software is thinking it's then when it isn't?  I haven't used reentrancy for about 4 or 5 deployments n verifications without issue i thought

**Comment by liarco** [comment]
Hi @negaunee042 code customizations might break the project. I need a complete reproducer in order to help you, this means:
1. you should provide me with a repository containing all of your code
2. you should tell me all the steps I have to do in order to reproduce your issue

If you can't share this or if your customizations are too complex, then it may be better looking for dedicated help on the Discord server.

Thank you 😉

**Comment by liarco** [comment]
@goshaminsk https://hardhat.org/guides/compile-contracts.html#artifacts

> Also, do you know what is contract library address?

Sorry but I don't understand the question, can you explain what you mean?

I'm gonna close this since I didn't receive any feedback in days.

**Comment by negaunee042** [comment]
> @goshaminsk https://hardhat.org/guides/compile-contracts.html#artifacts
> 
> > Also, do you know what is contract library address?
> 
> Sorry but I don't understand the question, can you explain what you mean?
> 
> I'm gonna close this since I didn't receive any feedback in days.

That's fine that you closed it.  I'm sorry I got sick and have been offline for several days.  I probably won't be back in full swing for a while, I have covid.  I appreciate the help tho.  If I can't figure it out when I get back in the swing I'll just start over.  Somewhere in the verification it's trying to verify Reentrancy when it was not in the the compiled/deployed contract.  I went file to file trying to locate where the problem is coming from, but didn't have luck.  I may have used the wrong encrypted abi for manual verification, I really don't know.  It's all good tho.  I'll take it as a lesson learned.  I really appreciate your offer tho, I know how busy you are.
